SYNC.md

```diff
@@ -0,0 +1,143 @@
+# Automerge Sync Protocol Documentation
+
+The automerge network sync protocol is designed to bring two documents into sync by exchanging messages between peers until both documents they have the same contents. It does this by exchanging rounds of sync messages. These sync messages contain two parts: 
+ * a lossily-compressed list of changes it already has (implicitly requesting the remainder)
+ * changes it believe the other peer needs
+
+Upon receiving a message, the developer calls `receiveSyncMessage(doc, syncState, message)`. This causes Automerge to apply any changes it received, then updates some internal metadata about the other peer and is ready to produce a reply. The developer can now call `generateSyncMessage(doc, syncState)` to produce the next message to a peer. These two functions are the main two functions and can continue to be called iteratively until `generateSyncMessage()` returns a `null` value indicating both documents are synchronized.
```

Comment:
Where does the `syncState` come from?

Reply:
Is it also worth mentioning when, and how to, get the sync process started here briefly (calling generate when a local change or another sync generated new changes)?